ksql-parser/src/main/java/io/confluent/ksql/parser/AstBuilder.java

```diff
@@ -126,1232 +130,1243 @@
+        case SqlBaseLexer.MINUS:
+          return ArithmeticUnaryExpression.negative(getLocation(context), child);
+        case SqlBaseLexer.PLUS:
+          return ArithmeticUnaryExpression.positive(getLocation(context), child);
+        default:
+          throw new UnsupportedOperationException(
+              "Unsupported sign: " + context.operator.getText());
       }
+    }
 
-      throw new InvalidColumnReferenceException(
-          columnLocation,
-          "Field " + columnName + " is ambiguous."
+    @Override
+    public Node visitArithmeticBinary(final SqlBaseParser.ArithmeticBinaryContext context) {
+      return new ArithmeticBinaryExpression(
+          getLocation(context.operator),
+          getArithmeticBinaryOperator(context.operator),
+          (Expression) visit(context.left),
+          (Expression) visit(context.right)
       );
     }
 
-    final Expression baseExpression =
-        new QualifiedNameReference(
-            getLocation(context),
-            QualifiedName.of(dataSourceExtractor.getFromAlias())
-        );
-    return new DereferenceExpression(getLocation(context), baseExpression, columnName);
-  }
+    @Override
+    public Node visitConcatenation(final SqlBaseParser.ConcatenationContext context) {
+      return new FunctionCall(
+          getLocation(context.CONCAT()),
+          QualifiedName.of("concat"), ImmutableList.of(
+          (Expression) visit(context.left),
+          (Expression) visit(context.right)
+      )
+      );
+    }
 
-  private boolean isValidNameOrAlias(final String name) {
-    // If this is join.
-    if (dataSourceExtractor.getJoinLeftSchema() != null) {
-      final boolean sameAsLeft = name.equalsIgnoreCase(dataSourceExtractor.getLeftAlias())
-          || name.equalsIgnoreCase(dataSourceExtractor.getLeftName());
-      final boolean sameAsRight = name.equalsIgnoreCase(dataSourceExtractor.getRightAlias())
-          || name.equalsIgnoreCase(dataSourceExtractor.getRightName());
-      return sameAsLeft || sameAsRight;
-    }
-    return ((name.equalsIgnoreCase(dataSourceExtractor.getFromAlias())
-        || name.equalsIgnoreCase(dataSourceExtractor.getFromName())));
-  }
+    @Override
+    public Node visitTimeZoneString(final SqlBaseParser.TimeZoneStringContext context) {
+      return new StringLiteral(
+          getLocation(context),
+          ParserUtil.unquote(context.STRING().getText(), "'")
+      );
+    }
 
-  @Override
-  public Node visitSimpleCase(final SqlBaseParser.SimpleCaseContext context) {
-    return new SimpleCaseExpression(
-        getLocation(context),
-        (Expression) visit(context.valueExpression()),
-        visit(context.whenClause(), WhenClause.class),
-        visitIfPresent(context.elseExpression, Expression.class)
-    );
-  }
+    @Override
+    public Node visitParenthesizedExpression(
+        final SqlBaseParser.ParenthesizedExpressionContext context) {
+      return visit(context.ex
```

Comment:
nit: argument should be on next line

Reply:
which? the `context.expression()`?  Personally, I only spread them one-per-line if they don't all fit on one line. one-per-line IMHO can lead to long methods that don't fit on one screen.

Reply:
I meant it seemed weird that the first argument was inline with the return-type and method name while the second argument was on its on line. Don't worry about it though.